waspc/src/Wasp/Generator/ServerGenerator/EmailG.hs

```diff
@@ -0,0 +1,143 @@
+    dstPath = C.serverSrcDirInServerRootDir </> [relfile|email/core/helpers.ts|]
+    tmplData =
+      object
+        [ "senderDefaults"
+            .= object
+              [ "email" .= AS.EmailSender.email emailFrom,
+                "title" .= title,
+                "isTitleDefined" .= isJust title
+              ]
+        ]
+    emailFrom = AS.EmailSender.defaultFrom email
+    title = AS.EmailSender.title emailFrom
+
+genSmtp :: EmailSender -> Generator [FileDraft]
+genSmtp email =
+  case AS.EmailSender.provider email of
+    AS.EmailSender.SMTP ->
+      sequence
+        [ copyTmplFile [relfile|email/core/providers/smtp.ts|]
+        ]
+    _ -> return []
+
+genSendGrid :: EmailSender -> Generator [FileDraft]
+genSendGrid email =
+  case AS.EmailSender.provider email of
+    AS.EmailSender.SendGrid ->
+      sequence
+        [ copyTmplFile [relfile|email/core/providers/sendgrid.ts|]
+        ]
+    _ -> return []
+
+genMailgun :: EmailSender -> Generator [FileDraft]
+genMailgun email =
+  case AS.EmailSender.provider email of
+    AS.EmailSender.Mailgun ->
+      sequence
+        [ copyTmplFile [relfile|email/core/providers/mailgun.ts|]
+        ]
+    _ -> return []
+
+getEmailProvidersJson :: EmailSender -> Aeson.Value
+getEmailProvidersJson email =
+  object
+    [ "isSmtpProviderUsed" .= isSmtpProviderUsed,
+      "isSendGridProviderUsed" .= isSendGridProviderUsed,
+      "isMailgunProviderUsed" .= isMailgunProviderUsed
+    ]
+  where
+    isSmtpProviderUsed = provider == AS.EmailSender.SMTP
+    isSendGridProviderUsed = provider == AS.EmailSender.SendGrid
+    isMailgunProviderUsed = provider == AS.EmailSender.Mailgun
+    provider = AS.EmailSender.provider email
+
+copyTmplFile :: Path' (Rel C.ServerTemplatesSrcDir) File' -> Generator FileDraft
+copyTmplFile = return . C.mkSrcTmplFd
+
+-- Dependencies
+
+nodeMailerVersionRange :: SV.Range
+nodeMailerVersionRange = SV.Range [SV.backwardsCompatibleWith (SV.Version 6 9 1)]
+
+nodeMailerDependency :: AS.Dependency.Dependency
+nodeMailerDependency = AS.Dependency.make ("nodemailer", show nodeMailerVersionRange)
+
+sendGridVersionRange :: SV.Range
+sendGridVersionRange = SV.Range [SV.backwardsCompatibleWith (SV.Version 7 7 0)]
+
+sendGridDependency :: AS.Dependency.Dependency
+sendGridDependency = AS.Dependency.make ("@sendgrid/mail", show sendGridVersionRange)
+
+mailgunVersionRange :: SV.Range
+mailgunVersionRange = SV.Range [SV.backwardsCompatibleWith (SV.Version 0 5 1)]
+
+mailgunDependency :: AS.Dependency.Dependency
+mailgunDependency = AS.Dependency.make ("ts-mailgun", show mailgunVersionRange)
+
+depsRequiredByEmail :: AppSpec -> [AS.Dependency.Dependency]
+depsRequiredByEmail spec =
+  concat
+    [ [nodeMailerDependency | provider == Just AS.EmailSender.SMTP],
+      [sendGridDependency | provider == Just AS.EmailSender.SendGrid],
+      [mailgunDependency | provider == Just AS.EmailSender.Mailgun]
+    ]
```

Comment:
One thing I think we can improve about this list of lists is that you already know it will always be just one of these, but it looks like it might be a combo of them. So I would rather we be explicit about that, and do a `case` that cases on provider and based on that returns correct dependency. Then I am reading the code and it is very clear what the output can be, while it looks like it could be a combo, but it can't. You will also get for free compiler warning when new provider is added, since it will be aware that that case wasn't handled in the `case`, while here you don't have that.